### PR TITLE
LEAN-2089

### PR DESCRIPTION
**Pull request by mnatsheh**
This PR looks big, but in reality is quite simple:
- Add general-purpose icons from assets.
- Remove unused icons defined in style-guide.
- Update icons to use consistent naming/props.
- Add icons story.

### CODE REVIEW

File: src/components/icons/edit.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*!
- * © 2019 Atypon Systems LLC
+ * © 2024 Atypon Systems LLC
```
Comment (mbartenev-atypon) [inline src/components/icons/edit.tsx:2]: why is this changed only in some file and in some it's not?

File: src/components/icons/plus.tsx [hunk 2]
```diff
@@ -18,30 +18,24 @@ import React from 'react'
...
     xmlns="http://www.w3.org/2000/svg"
-    transform={transform}
+    {...props}
   >
     <path
       fillRule="evenodd"
       clipRule="evenodd"
       d="M5.5 0.5C5.25147 0.5 5.05 0.701472 5.05 0.95V4.55H1.45C1.20147 4.55 1 4.75147 1 5C1 5.24853 1.20147 5.45 1.45 5.45H5.05V9.05C5.05 9.29853 5.25147 9.5 5.5 9.5C5.74853 9.5 5.95 9.29853 5.95 9.05V5.45H9.55C9.79853 5.45 10 5.24853 10 5C10 4.75147 9.79853 4.55 9.55 4.55H5.95V0.95C5.95 0.701472 5.74853 0.5 5.5 0.5Z"
-      fill={color}
+      fill="#6E6E6E"
```
Comment (mbartenev-atypon) [inline src/components/icons/plus.tsx:34]: why do we not want components like that to be adjustable in terms of colors and sizes?

File: src/components/icons/edit.tsx (hunk 1 above)
Comment (mnatsheh) [reply src/components/icons/edit.tsx:2]: This is actually a new file, but for some reason git is seeing it as a rename of another file that I deleted "bookmark".

File: src/components/icons/plus.tsx (hunk 2 above)
Comment (mnatsheh) [reply src/components/icons/plus.tsx:34]: Most of this should be possible using the passed `props`, and the rest can be done with styled-components. I don't think there is a need for some icons to work differently.

**Review by mbartenev-atypon — APPROVED** [review]
